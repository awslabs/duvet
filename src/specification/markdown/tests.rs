// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

use super::*;

fn tokens(contents: &str) -> Vec<Token> {
    Lex::new(contents).collect()
}

macro_rules! snapshot {
    ($name:ident, $contents:expr) => {
        mod $name {
            #[test]
            fn tokens() {
                insta::assert_debug_snapshot!("tokens", super::tokens($contents));
            }

            #[test]
            fn tree() {
                insta::assert_debug_snapshot!("tree", super::parse($contents));
            }
        }
    };
}

snapshot!(
    simple,
    r#"
# This is a test

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multi_line_header,
    r#"
Foo *bar
baz*
======

Content goes here. Another
sentence here.
"#
);

snapshot!(
    multiple,
    r#"
# This is a test

Content goes here. Another
sentence here.

## This is another test

More content goes here

### Nested section

Testing 123

## Up one

Another section
"#
);

snapshot!(
    list_example,
    r#"
# List example

Here is a list:
* Item 1
* Item 2
  * Item 2.1
* Item 3
  * Item 3.1
    * Item 3.1.1
    * Item 3.1.2
  * Item 3.2

Here is a numbered list:
1. Item 1
2. Item 2
3. Item 3

Here is a list with content:
* Item
  More content

  Other content

  * Testing

    Other test

Testing 123

* Item
More content
"#
);
