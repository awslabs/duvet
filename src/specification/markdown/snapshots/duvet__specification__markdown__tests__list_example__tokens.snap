---
source: src/specification/markdown/tests.rs
expression: "super::tokens(r#\"\n# List example\n\nHere is a list:\n* Item 1\n* Item 2\n  * Item 2.1\n* Item 3\n  * Item 3.1\n    * Item 3.1.1\n    * Item 3.1.2\n  * Item 3.2\n\nHere is a numbered list:\n1. Item 1\n2. Item 2\n3. Item 3\n\nHere is a list with content:\n* Item\n  More content\n\n  Other content\n\n  * Testing\n\n    Other test\n\nTesting 123\n\n* Item\nMore content\n\"#)"
---
[
    Line(
        Str {
            value: "",
            pos: 0,
            line: 1,
        },
    ),
    Header {
        line: Str {
            value: "# List example",
            pos: 1,
            line: 2,
        },
        name: Str {
            value: "List example",
            pos: 3,
            line: 2,
        },
        fragment: None,
        level: 1,
    },
    Line(
        Str {
            value: "",
            pos: 16,
            line: 3,
        },
    ),
    Line(
        Str {
            value: "Here is a list:",
            pos: 17,
            line: 4,
        },
    ),
    Break,
    Line(
        Str {
            value: "* Item 1",
            pos: 33,
            line: 5,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "* Item 2",
            pos: 42,
            line: 6,
        },
    ),
    Break,
    Line(
        Str {
            value: "  * Item 2.1",
            pos: 51,
            line: 7,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "* Item 3",
            pos: 64,
            line: 8,
        },
    ),
    Break,
    Line(
        Str {
            value: "  * Item 3.1",
            pos: 73,
            line: 9,
        },
    ),
    Break,
    Line(
        Str {
            value: "    * Item 3.1.1",
            pos: 86,
            line: 10,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "    * Item 3.1.2",
            pos: 103,
            line: 11,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "  * Item 3.2",
            pos: 120,
            line: 12,
        },
    ),
    Break,
    Line(
        Str {
            value: "",
            pos: 133,
            line: 13,
        },
    ),
    Line(
        Str {
            value: "Here is a numbered list:",
            pos: 134,
            line: 14,
        },
    ),
    Break,
    Line(
        Str {
            value: "1. Item 1",
            pos: 159,
            line: 15,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "2. Item 2",
            pos: 169,
            line: 16,
        },
    ),
    Break,
    Break,
    Line(
        Str {
            value: "3. Item 3",
            pos: 179,
            line: 17,
        },
    ),
    Break,
    Line(
        Str {
            value: "",
            pos: 189,
            line: 18,
        },
    ),
    Line(
        Str {
            value: "Here is a list with content:",
            pos: 190,
            line: 19,
        },
    ),
    Break,
    Line(
        Str {
            value: "* Item",
            pos: 219,
            line: 20,
        },
    ),
    Line(
        Str {
            value: "  More content",
            pos: 226,
            line: 21,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 241,
            line: 22,
        },
    ),
    Line(
        Str {
            value: "  Other content",
            pos: 242,
            line: 23,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 258,
            line: 24,
        },
    ),
    Break,
    Line(
        Str {
            value: "  * Testing",
            pos: 259,
            line: 25,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 271,
            line: 26,
        },
    ),
    Line(
        Str {
            value: "    Other test",
            pos: 272,
            line: 27,
        },
    ),
    Break,
    Line(
        Str {
            value: "",
            pos: 287,
            line: 28,
        },
    ),
    Line(
        Str {
            value: "Testing 123",
            pos: 288,
            line: 29,
        },
    ),
    Line(
        Str {
            value: "",
            pos: 300,
            line: 30,
        },
    ),
    Break,
    Line(
        Str {
            value: "* Item",
            pos: 301,
            line: 31,
        },
    ),
    Line(
        Str {
            value: "More content",
            pos: 308,
            line: 32,
        },
    ),
    Break,
]
